test/GithubApi.Repositories.test.js

```diff
@@ -0,0 +1,46 @@
+const axios = require('axios');
+const { expect } = require('chai');
+const { StatusCodes } = require('http-status-codes');
+
+const urlBase = 'https://api.github.com/users';
+const urlRepos = 'https://api.github.com/repos';
+const repo = 'jasmine-json-report';
+const owner = 'aperdomob';
+
+describe('Repositories testing', () => {
+  it('GET method', async () => {
+    const response = await axios.get(`${urlBase}/${owner}`);
+
+    expect(response.status).to.equal(StatusCodes.OK);
+    expect(response.data.name).to.equal('Alejandro Perdomo');
+    expect(response.data.company).to.equal('Perficient Latam');
+    expect(response.data.location).to.equal('Colombia');
+  });
+
+  it('Repositories List', async () => {
+    const response = await axios.get(`${urlBase}/${owner}/repos`);
+    const finder = response.data.find((object) => object.name === 'jasmine-json-report');
+
+    expect(response.status).to.equal(StatusCodes.OK);
+    expect(finder.name).to.equal('jasmine-json-report');
+    expect(finder.private).to.equal(false);
+    expect(finder.description).to.equal('A Simple Jasmine JSON Report');
+  });
+
+  it('Repo download', async () => {
+    const zipFile = await axios.get(`${urlRepos}/${owner}/${repo}/zipball`);
```

Comment:
In the previous test, you are getting the repository with the find expression from the response, you can declare that variable as global and use it here.  to download the zip in this way:
```
.get(`${repository.svn_url}/archive/${repository.default_branch}.zip`
```

Reply:
I'm having troubles declaring the variable as global in next scope because it has another variable inside, defined in other instances. I already downloaded the zip file. Is totally necessary to perform this change?

Reply:
yes, take a look at my new comments for an idea on how to solve this.